packages/flutter_tools/gradle/src/main/groovy/app_plugin_loader.groovy

```diff
@@ -0,0 +1,38 @@
+import groovy.json.JsonSlurper
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+
+apply plugin: FlutterAppPluginLoaderPlugin
+
+class FlutterAppPluginLoaderPlugin implements Plugin<Settings> {
+    @Override
+    void apply(Settings settings) {
+        def flutterProjectRoot = settings.settingsDir.parentFile
+
+        // If this logic is changed, also change the logic in module_plugin_loader.gradle.
+        def pluginsFile = new File(flutterProjectRoot, '.flutter-plugins-dependencies')
+        if (!pluginsFile.exists()) {
+            return
+        }
+
+        def object = new JsonSlurper().parseText(pluginsFile.text)
+        assert object instanceof Map
+        assert object.plugins instanceof Map
+        assert object.plugins.android instanceof List
+        // Includes the Flutter plugins that support the Android platform.
+        object.plugins.android.each { androidPlugin ->
+            assert androidPlugin.name instanceof String
+            assert androidPlugin.path instanceof String
+            // Skip plugins that have no native build (such as a Dart-only implementation
+            // of a federated plugin).
+            def needsBuild = androidPlugin.containsKey('native_build') ? androidPlugin['native_build'] : true
```

Comment:
Nit: use named constant here and link to documentation for where maintainers can read more about it.

Reply:
This is copy-pasted, but I'm more than happy to make improvements like this (after I get all the tests to pass...)

Reply:
nit: now that the tests pass lets use a constant here.

Reply:
> link to documentation for where maintainers can read more about it.

That's a good idea but I searched for a while but couldn't find any docs referring `native_build`. It's defined here:

https://github.com/flutter/flutter/blob/ff10c52ad6de098b4946f9ef33fdde8ebd5bc594/packages/flutter_tools/lib/src/flutter_plugins.dart#L106